### ISSUE TITLE
[Feature Suggestion] Add Undo Save/Load on the menu

### ISSUE THREAD

**Issue by capitaineflam25** [MiSTer-devel/Main_MiSTer#791]
Overwriting the current save state by mistake can sometimes be very frustrating when playing.
A file based undo feature could help solving that, like the one on RetroArch.

Assume you're playing SMB on the NES core, and current save slot is 1.

Before saving, SMB_1.ss can be renamed to SMB_1_prev.ss
If "Undo Save State" is pressed, SMB_1_prev.ss is then renamed to SMB_1.ss

Before restoring a state, current state can be saved to SMB_1_undo.ss
If "Undo Load State" is pressed, SMB_1_undo.ss can be loaded.

**Comment by sorgelig** [comment]
There are different ways to use save state to avoid frustration. For example you can use one slot for long-term savestate while other slot for temporary save state till you achieve required state and then save to long-term slot.

**Comment by capitaineflam25** [comment]
Thanks for the suggestion.

That's the kind of thing i'm actually doing to avoid overwriting by mistake my best save state.
It requires a bit of practice to master the technique 😁

**Comment by sorgelig** [comment]
this is good option too.

**Comment by birdybro** [comment]
Yeah, pretty much every emulator has instant save/load of savestates via hotkeys or controller button combinations, so this problem is not really endemic to MiSTer alone. I am looking at updating savestate_ui on the existing cores that use it to mirror the PSX core.
